deep-learning/src/main/python/synapse/ml/HuggingFaceSentenceEmbedder.py

```diff
@@ -0,0 +1,351 @@
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Import necessary libraries
+import numpy as np
+import torch
+import pyspark.sql.functions as F
+import tensorrt as trt
+import logging
+import warnings
+import sys
+import datetime
+import pytz
+from tqdm import tqdm, trange
+from numpy import ndarray
+from torch import Tensor
+from typing import List, Union
+
+import model_navigator as nav
+from sentence_transformers import SentenceTransformer
+from sentence_transformers.util import batch_to_device
+from pyspark.ml.functions import predict_batch_udf
+from faker import Faker
```

Comment:
can we remove this dep as previously discussed? you can add a little fake data passage and replicate it if you need